configs/components/_base-ruby-augeas.rb

```diff
@@ -74,6 +76,9 @@
   build_commands = []
   build_commands << "#{ruby} ext/augeas/extconf.rb"
   build_commands << "#{platform[:make]} -e -j$(shell expr $(shell #{platform[:num_cores]}) + 1)"
+  if ruby_version =~ /^3/
+    build_commands << "#{platform.patch} --strip=2 --fuzz=0 --ignore-whitespace --no-backup-if-mismatch < ../ruby-augeas-0.5.0-patch_c_extension.patch"
+  end
```

Comment:
I think you'll want to patch before running `make`, but I'm not sure if patching needs to come before `ruby ext/augeas/extconf.rb`?

Reply:
Ah okay, I'll move that. What would be the best way to check if I should patch before ruby ext/augeas/extconf.rb too?

Reply:
One option is to build the runtime using `--preserve always` and then ssh to the build host and run (I think):

```
$ /opt/puppetlabs/puppet/bin/ruby -e "require 'augeas'"
```

In theory that should be sufficient for the augeas ruby bindings to load the native library and trigger the warning.

That said, it looks we do all of the patching early in the `ruby-shadow` component (yet another rubygem with native extensions that we patch): https://github.com/puppetlabs/puppet-runtime/blob/b14bfccd97d13eead785d7aa0ca589de79a36623/configs/components/ruby-shadow.rb#L44-L57 So maybe follow that?

Reply:
Hmm `$ /opt/puppetlabs/puppet/bin/ruby -e "require 'augeas'"` doesn't trigger a warning on a host that doesn't have the patch